src/main/java/me/itzg/helpers/modrinth/ModrinthPackInstaller.java

```diff
@@ -0,0 +1,215 @@
+package me.itzg.helpers.modrinth;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import lombok.extern.slf4j.Slf4j;
+import me.itzg.helpers.errors.GenericException;
+import me.itzg.helpers.errors.InvalidParameterException;
+import me.itzg.helpers.fabric.FabricLauncherInstaller;
+import me.itzg.helpers.files.IoStreams;
+import me.itzg.helpers.forge.ForgeInstaller;
+import me.itzg.helpers.http.SharedFetchArgs;
+import me.itzg.helpers.json.ObjectMappers;
+import me.itzg.helpers.modrinth.model.*;
+import me.itzg.helpers.quilt.QuiltInstaller;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+@Slf4j
+public class ModrinthPackInstaller {
+    private final ModrinthApiClient apiClient;
+    private final Path zipFile;
+    private final Path outputDirectory;
+    private final Path resultsFile;
+    private final boolean forceModloaderReinstall;
+    private final SharedFetchArgs sharedFetchArgs;
+
+    public ModrinthPackInstaller(ModrinthApiClient apiClient, SharedFetchArgs sharedFetchArgs, Path zipFile, Path outputDirectory, Path resultsFile, boolean forceModloaderReinstall) {
```

Comment:
I was just thinking, rather pass in `SharedFetchArgs` it would be cleaner to pass in `SharedFetch.Options`.

Reply:
This part confuses me: the only consumer of this in the whole class is the Quilt loader installer; why does the Quilt loader need it while Forge and Fabric don't? (I acknowledge this question is for my curiosity and desire for architectural consistency; having spent so much time in this code now, i want to come back to this class and clean it up but that falls beyond the scope of what i am trying to achieve here)

Reply:
Fair question 🙂. I have been evolving the design over time, so Quilt is the newer of the three implementations with cleaner shared fetch usage. If you're interested in a little extra cleanup, the `SharedFetch.Options` should really be passed in rather than the default used here from the `Options.builder().build()` call.

https://github.com/itzg/mc-image-helper/blob/580e62b42f01cc38296e1abd221ec2737ff59da6/src/main/java/me/itzg/helpers/fabric/FabricLauncherInstaller.java#L48

Reply:
...and Forge installer is "pre shared fetch", so it's calling `me.itzg.helpers.http.Fetch#fetch` to create one-off instances.